### ISSUE TITLE
scaled_dot_product_attention output is different from memory_efficient_attention

### ISSUE THREAD

**Issue by aenoca** [facebookresearch/xformers#1114]
# ❓ Questions and Help
I have been playing with DINOv2-small and trying to replicate some of its outputs by rebuilding it from scratch. I noticed that the results I got when running with scaled_dot_product_attention versus with memory_efficient_attention were NOT the same. 

I wonder if anyone can check this out.

**Comment by sebhtml** [comment]
Can you provide more details including:
- small code example that reproduce the problem,
- expected result
- actual result
?

**Comment by huangmz21** [comment]
Actually,the results provided by xformers are also slightly different from the results I got by the standard Attention wrote by myself.Code is here:
def normalized_attention(query, key, value, attn_bias=None, scale=None, p=0.0):
    """

    """
    


    query = query.permute(0,2,3,1,4)
    key = key.permute(0,2,3,1,4)
    value = value.permute(0,2,3,1,4)
    attn_scores = torch.matmul(query, key.transpose(-2, -1))  # (batch_size, num_heads, seq_len_q, seq_len_k)

    if attn_bias is not None:
        attn_scores = attn_scores + attn_bias


    if scale is not None:
        attn_scores = attn_scores * scale

    attn_probs = F.softmax(attn_scores, dim=-1)

    if p > 0.0:
        attn_probs = F.dropout(attn_probs, p=p, training=True)


    out = torch.matmul(attn_probs, value)  # (batch_size, num_heads, seq_len_q, dim_v)
    out=out.permute(0,3,1,2,4)

    return out  ,                

out = xops.memory_efficient_attention_forward(
                        query,
                        key,
                        value,
                        attn_bias=cache_fuse_metadata["attn_bias"],
                        p=0.0,
                        scale=self.scale,
                    ),out_hmz = normalized_attention(query, key, value, attn_bias=cache_fuse_metadata["attn_bias"], scale=self.scale, p=0.0)  and the diff like this :torch.sum(abs(out-out_hmz))
tensor(0.5000, device='cuda:0', dtype=torch.bfloat16),out.shape
torch.Size([1, 126, 8, 4, 128])
out_hmz.shape
torch.Size([1, 126, 8, 4, 128])
